txstratum/prometheus.py

```diff
@@ -16,21 +26,59 @@
 METRIC_INFO: Dict[str, str] = {
     'miners_count': 'Number of connected miners',
     'total_hashrate_ghs': 'Hashrate (Gh/s)',
-    'txs_solved': 'Number of solved transactions',
     'txs_timeout': 'Number of timeouts when solving transactions',
     'blocks_found': 'Number of blocks found',
     'uptime': 'Service uptime',
     'tx_queue': 'Number of transactions in the queue',
     'block_template_error': 'Number of errors updating block template',
 }
 
+# Metrics that we update through pubsub events
+METRICS_PUBSUB = {
+    'txs_jobs_received': Counter(
+        'txs_jobs_received',
+        'Number of transactions received by the API',
+    ),
+    'txs_solved': Counter(
+        'txs_solved',
+        'Number of solved transactions',
+        labelnames=['miner_type', 'miner_address'],
+    ),
+    'txs_solved_weight': Histogram(
+        'txs_solved_weight',
+        'Txs solved histogram by tx weight',
+        buckets=(17, 18, 19, 20, 21, 22, 23, 25, float("inf")),
+        labelnames=['miner_type']
+    ),
+    'txs_timeout_weight': Histogram(
+        'txs_timeout_weight',
+        'Txs timeouts histogram by tx weight',
+        buckets=(17, 18, 19, 20, 21, 22, 23, 25, float("inf"))
```

Comment:
Why no 24?

Reply:
Same reason as in https://github.com/HathorNetwork/tx-mining-service/pull/54#discussion_r787137761